src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -199,7 +199,10 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     catch (Exception ex)
                     {
                         _syncAfter = DateTimeUtil.Add(now.UtcDateTime, _automaticRefreshInterval < _refreshInterval ? _automaticRefreshInterval : _refreshInterval);
-                        throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX20803, (_metadataAddress ?? "null")), ex));
+                        if (_currentConfiguration == null) // Throw an exception if there's no configuration to return.
+                            throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX20803, (_metadataAddress ?? "null")), ex));
```

Comment:
>throw LogHelper. [](start = 28, length = 16)

Consider not throwing and logging here, this is happening further down with a similar message.

Reply:
I would like to continue to throw here as we could provide details that may not be available later.
I am suggesting we throw / inform users to understand metadata retrieval failed.

Reply:
I was considering only throwing further down, but this will cause us to lose information about the original exception that was thrown.

Reply:
That information can be stored for later usage